google/cloud/storage/tests/testbench.py

```diff
@@ -1238,6 +1262,58 @@ def objects_update(bucket_name, object_name):
     return json.dumps(revision.metadata)
 
+@gcs.route('/b/<bucket_name>/o/<object_name>/compose', methods=['POST'])
+def objects_compose(bucket_name, object_name):
+    """Implement the 'Objects: compose' API: concatenate Objects."""
+    composed_object_path = bucket_name + '/o/' + object_name
+    gcs_object = GCS_OBJECTS.get(composed_object_path,
+                                 GcsObject(bucket_name, object_name))
+    if gcs_object is not None:
+        gcs_object.check_preconditions(flask.request)
+    payload = json.loads(flask.request.data)
+    source_objects = payload["sourceObjects"] 
+    if source_objects is None:
+        raise ErrorResponse('You must provide at least one source component.',
+                            status_code=400)
+    if len(source_objects) > 32:
+        raise ErrorResponse('The number of source components provided'
+                            ' (%d) exceeds the maximum (32)' %
+                                len(source_objects), status_code=400)
+    composed_media = ""
+    for source_object in source_objects:
+        source_object_name = source_object.get('name')
+        if source_object_name is None:
+            raise ErrorResponse('Required.', status_code=400)
+        source_object_path = bucket_name + '/o/' + source_object_name
+        source_gcs_object = GCS_OBJECTS.get(source_object_path,
+                                           GcsObject(bucket_name,
+                                                     source_object_name))
+        if source_gcs_object is None:
+            raise ErrorResponse('No such object: %s' % source_object_path,
+                                status_code=404)
+        source_revision = source_gcs_object.get_latest()
+        generation = source_object.get('generation')
+        if generation is not None:
+            source_revision = source_gcs_object.get_revision_by_generation(
+                generation)
+            if source_revision is None:
+                raise ErrorResponse('No such object: %s' % source_object_path,
+                                    status_code=404)
+        object_preconditions = source_object.get('objectPreconditions')
+        if object_preconditions is not None:
+            if_generation_match = object_preconditions.get('ifGenerationMatch')
+            if if_generation_match is not None:
+                if source_gcs_object.generation != if_generation_match:
+                    raise ErrorResponse('Precondition Failed', status_code=412)
+        composed_media += source_revision.media
+    gcs_object = GCS_OBJECTS.setdefault(composed_object_path,
+                                        GcsObject(bucket_name, object_name))
+    GCS_OBJECTS[composed_object_path] = gcs_object
+    base_url = flask.url_for('gcs_index', _external=True)
+    current_version = gcs_object.compose_from(base_url, flask.request,
+        composed_media)
+    return json.dumps(current_version.metadata)
```

Comment:
Please use `filtered_response(flask.request, current_version.metadata)`.  I think we have a few places where we are not using `filtered_response()`, those are bugs, I probably missed them when we merged many changes more or less at the same time.